### PR TITLE
feat: burning crusade classic

### PR DESCRIPTION
**Pull request by casperstorm**
This PR will add support for Burning Crusade.

Currently we do not have enough information from Blizzard to know which naming they are choosing for the Burning Crusade, so until this is known, this PR will be blocked:

Mostly a TODO for myself.

- [x] Add _classic_beta_
  - [x] Hub
  - [x] Curse
- [x] Classic will be renamed from _classic_ to _classic_era_
- [x] Burning Crusade will use _classic_
  - [x] Hub
  - [x] Curse
  - [x] Create it as a new base flavor  

## Checklist

- [ ] Tested on Windows
- [ ] Tested on MacOS
- [ ] Tested on Linux
- [ ] Added an entry to `CHANGELOG.md` if knowledge of this change could be valuable to users

### CODE REVIEW

**Comment by fleps** [comment]
Does the current last release support TBC Beta in any way?
I've installed and I can see my TBC as "Classic Beta" on the path, but the catalog / install from Curse still uses / install the Classic version of the addons, not the TBC.

**Comment by casperstorm** [comment]
> Does the current last release support TBC Beta in any way?
> 
> I've installed and I can see my TBC as "Classic Beta" on the path, but the catalog / install from Curse still uses / install the Classic version of the addons, not the TBC.

Not yet. We are awaiting Blizzard to expose folder structure to us. Also Curse API needs to update which has still hot happened.

**Comment by joaomcarlos** [comment]
Looking forward to this. Everyones waiting :D

File: crates/core/src/config/wow.rs [hunk 1]
```diff
@@ -34,18 +34,21 @@ pub enum Flavor {
     #[serde(alias = "RetailPTR")]
     RetailPtr,
     RetailBeta,
-    #[serde(alias = "classic", alias = "wow_classic")]
+    #[serde(rename = "classic", rename = "wow_classic", rename = "classic_era")]
```
Comment (casperstorm) [inline crates/core/src/config/wow.rs:37]: @tarkah seems like we can't multiple `rename`.

**Pull request merged**